validation.go

```diff
@@ -27,6 +24,29 @@ var ErrBadRecord = errors.New("bad dht record")
 // is not found in the Validator map of the DHT.
 var ErrInvalidRecordType = errors.New("invalid record keytype")
 
+type ValidationRecord struct {
+  key string
+  value []byte
+  author *peer.ID
```

Comment:
it's a little odd to have a pointer to what is a typedef to a string.
Perhaps we should just compare with the empty string instead of nil, and then we won't have to dereference a pointer.

Reply:
I agree it is a little strange. I was trying to decide what was stranger between that and `peer.ID("")`

Reply:
I think we can just compare with the string literal `""`, without the explicit casting, and it should work.